cinema/serializers.py

```diff
@@ -68,13 +79,81 @@
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionSerializer()
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "movie_session", "row", "seat")
+
+    def validate(self, attrs):
+        movie_session = attrs.get("movie_session")
+        row = attrs.get("row")
+        seat = attrs.get("seat")
+        cinema_hall = movie_session.cinema_hall
```

Comment:
The `cinema_hall` attribute is accessed directly from `movie_session`. Ensure that `movie_session` is a fully populated instance, or adjust the logic to handle cases where it might not be.